src/Raven.Server/Documents/Indexes/Persistence/Corax/CoraxIndexReadOperation.cs

```diff
@@ -55,66 +60,114 @@ public CoraxIndexReadOperation(Index index, Logger logger, Transaction readTrans
+                    {
+                        docsToLoad++;
+                        skippedResults.Value++;
+                        continue;
+                    }
+
+                    var retrieverInput = new RetrieverInput(_fieldMappings, _indexSearcher.GetReaderAndIdentifyFor(longIds[i], out var key), key);
+                    bool markedAsSkipped = false;
                     var fetchedDocument = retriever.Get(ref retrieverInput, token);
 
                     if (fetchedDocument.Document != null)
                     {
-                        yield return new QueryResult {Result = fetchedDocument.Document};
+                        var qr = GetQueryResult(fetchedDocument.Document, ref markedAsSkipped);
+                        if (qr.Result is null)
+                        {
+                            docsToLoad++;
+                            continue;
+                        }
+
+                        yield return qr;
                     }
                     else if (fetchedDocument.List != null)
                     {
                         foreach (Document item in fetchedDocument.List)
                         {
-                            yield return new QueryResult {Result = item};
+                            var qr = GetQueryResult(item, ref markedAsSkipped);
+                            if (qr.Result is null)
+                            {
+                                docsToLoad++;
+                                continue;
+                            }
+
+                            yield return qr;
                         }
                     }
                 }
 
-                if (docsToLoad == 0)
+                if ((read = queryMatch.Fill(longIds)) == 0)
                     break;
-
-                read = result.Fill(ids);
-                readCounter += read;
+                totalResults.Value += read;
             }
 
-            if (result is SortingMatch sm)
-                totalResults.Value = Convert.ToInt32(sm.TotalResults);
-            else
+            if (isDistinctCount)
+                totalResults.Value -= skippedResults.Value;
+
+            QueryResult GetQueryResult(Document document, ref bool markedAsSkipped)
             {
-                while (read != 0)
+                if (queryScope.TryIncludeInResults(document) == false)
                 {
-                    read = result.Fill(ids);
-                    readCounter += read;
+                    document?.Dispose();
+
+                    if (markedAsSkipped == false)
+                    {
+                        skippedResults.Value++;
+                        markedAsSkipped = true;
+                    }
+
+                    return default;
+                }
+
+                if (isDistinctCount == false)
+                {
+                    Dictionary<string, Dictionary<string, string[]>> highlightings = null;
```

Comment:
Check the warning - `The variable 'highlightings' is assigned but its value is never used`

Reply:
Deleted